xdsl/tools/xdsl-opt

```diff
@@ -15,6 +17,43 @@ class OptMain(xDSLOptMain):
     def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
         super().register_all_arguments(arg_parser)
 
+        #arg_parser.add_argument(
+        #    "--check-coverage",
+        #    default=False,
+        #    action='store_true',
+        #    help="Generate the xDSL code coverage for this run.")
+
+    def run(self):
+        # In order to find the `.coverage` file, as well as the `.coveragerc`,
+        # we need to step up to the `xdsl` directory. Notice that lit runs tests
+        # within the folder their are contained in.
```

Comment:
This is a very weird issue and comes from the fact that `coverage` is launched in different subdirectories when using lit. This means that `.coveragerc` is sometimes not found or in different relative paths. Does anyone have a cleaner idea how to implement this?

Reply:
Can we give as option to `xdsl-opt` the link to `.coveragerc` ?

Reply:
The problem is that this cannot be an absolute path, since those are different on different machines, but the relative paths are always different since the program runs in different directories in each lit run.

Another potential solution is trying to configure lit in a way that it actually executes everything in the root directory. Then this would be unnecessary as well.